### PR TITLE
{devel}[intel/2018b] Spark 2.4.0 w/ Python 2.7.15 & 3.6.6 + Hadoop 2.9.2 (+ deps)

### PR DESCRIPTION
**Pull request by boegel**
(created using `eb --new-pr`)

requires ~~#7479~~ (Arrow)

### CODE REVIEW

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 5 out of 5 (5 easyconfigs in this PR)
node2097.delcatty.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/eccbade7020134d149d1843a66cd3ebc for a full test report.

**Comment by vanzod** [comment]
Test report by @vanzod
**FAILED**
Build succeeded for 2 out of 6 (5 easyconfigs in this PR)
cermis - Linux debian 9.6, Intel(R) Xeon(R) CPU           E5620  @ 2.40GHz, Python 2.7.13
See https://gist.github.com/53b40bf1d5ceaf5250b8db35be83f2e6 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 6 out of 6 (6 easyconfigs in this PR)
node2404.golett.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/954bef6f2de03471d3c3202cefa4c404 for a full test report.

File: easybuild/easyconfigs/p/protobuf/protobuf-2.5.0-GCCcore-7.3.0.eb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+name = 'protobuf'
+version = '2.5.0'
+
+homepage = 'https://github.com/google/protobuf/'
+description = """Google Protocol Buffers"""
+
+toolchain = {'name': 'GCCcore', 'version': '7.3.0'}
+
+source_urls = ['https://github.com/google/protobuf/archive/v%(version)s/']
```
Comment (vanzod) [inline easybuild/easyconfigs/p/protobuf/protobuf-2.5.0-GCCcore-7.3.0.eb]: The checksum of the archive from this URL is different from the one downloaded from `https://github.com/google/%(namelower)s/releases/download/v%(version)s/`

**Comment by vanzod** [comment]
Test report by @vanzod
**FAILED**
Build succeeded for 3 out of 6 (6 easyconfigs in this PR)
cermis - Linux debian 9.6, Intel(R) Xeon(R) CPU           E5620  @ 2.40GHz, Python 2.7.13
See https://gist.github.com/4042b9bbd0800e480969cc8135b434db for a full test report.

**Comment by vanzod** [comment]
@boegel `zlib` is missing as dependency in `Hadoop-2.9.2-GCCcore-7.3.0-native.eb`

**Comment by boegel** [comment]
@vanzod `zlib` is there as a build dep for `Hadoop`, but maybe it needs to be a runtime dep?

**Comment by vanzod** [comment]
@boegel Here you can find the full log:
https://gist.github.com/vanzod/47c8b0862d3f15c759315cdfe0b7fa38

**Comment by boegel** [comment]
@vanzod Here's the problem:

```
[WARNING] CMake Error at /opt/easybuild/software/Compiler/GCCcore/7.3.0/CMake/3.12.1/share/cmake-3.12/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
[WARNING]   Could NOT find ZLIB (missing: ZLIB_INCLUDE_DIR)
```

Not sure why that happens though, since the location of the provided `zlib` is specified to the build command via ` -Dzlib.prefix=/opt/easybuild/software/Compiler/GCCcore/7.3.0/zlib/1.2.11`...

I don't see that being passed down to the `cmake` command though, while it does happen for `snappy` (see ` -DCUSTOM_SNAPPY_PREFIX=...`)

**Comment by boegel** [comment]
@vanzod Can you check if https://github.com/easybuilders/easybuild-easyblocks/pull/1787 resolve the problem for you?

**Comment by smoors** [comment]
Test report by @smoors
**SUCCESS**
Build succeeded for 7 out of 7 (6 easyconfigs in this PR)
node302.hydra.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 2.7.5
See https://gist.github.com/7e51cd152ff93a769a0ab333d4ae2082 for a full test report.

**Comment by smoors** [comment]
(this test report was without applying easybuilders/easybuild-easyblocks#1787)

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
node2412.golett.os - Linux centos linux 7.7.1908, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/72902162f04403a5ccf7bd2b60fb9904 for a full test report.

**Comment by boegel** [comment]
I managed to construct a CentOS 7 container image in which `zlib-devel` is missing (removed via `rpm -e --nodeps zlib-devel`), which allowed me to reproduce the problem @vanzod was seeing (see failing test report).

```
[WARNING] CMake Error at /software/CMake/3.12.1-GCCcore-7.3.0/share/cmake-3.12/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
[WARNING]   Could NOT find ZLIB (missing: ZLIB_INCLUDE_DIR)
[WARNING] Call Stack (most recent call first):
[WARNING]   /software/CMake/3.12.1-GCCcore-7.3.0/share/cmake-3.12/Modules/FindPackageHandleStandardArgs.cmake:378 (_FPHSA_FAILURE_MESSAGE)
[WARNING]   /software/CMake/3.12.1-GCCcore-7.3.0/share/cmake-3.12/Modules/FindZLIB.cmake:112 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
[WARNING]   CMakeLists.txt:47 (find_package)
```

Now I can try to figure out why the `zlib` dependency isn't being picked up...

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 6 out of 6 (6 easyconfigs in this PR)
node2412.golett.os - Linux centos linux 7.7.1908, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/b59074ed98a4c606544fa181c9cb9a84 for a full test report.

**Comment by Micket** [comment]
Going in, thanks @boegel!

**Pull request merged**